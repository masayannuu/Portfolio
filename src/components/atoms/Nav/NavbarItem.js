import React from 'react'
import styled from 'styled-components'

const NavAnker = styled.a`
  color: #ffffff;
  font-feature-settings: 'palt';
  letter-spacing: 0.2em;
  font-size: 81.25%;
  &:hover {
    color: #f0f0f0;
    text-decoration: underline;
    -webkit-transition: all 0.5s ease;
    -moz-transition: all 1s ease;
    text-shadow: #057D9F 0px 0px 20px;
  }
`

const NavbarItem = ({ children, href }) => (
  <div className="navbar-item">
    <p className="control">
      <NavAnker href={href}>{children}</NavAnker>
    </p>
  </div>
)

export default NavbarItem
