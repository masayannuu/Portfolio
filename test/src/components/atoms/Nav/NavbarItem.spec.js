import React from 'react'
import Enzyme from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import NavbarItem from '../../../../../src/components/atoms/Nav/NavbarItem'

describe('Nav NavbarItem Atoms', () => {
  it('should render NavbarItem', () => {
    const tag = (
      <NavbarItem href="hoge">
        <p>test</p>
      </NavbarItem>
    )
    const wrapper = Enzyme.shallow(tag)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
