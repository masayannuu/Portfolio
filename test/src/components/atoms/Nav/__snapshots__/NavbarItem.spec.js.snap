// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav NavbarItem Atoms should render NavbarItem 1`] = `
<div
  className="navbar-item"
>
  <NavbarItem__NavAnker
    href="hoge"
  >
    <span>
      <p>
        test
      </p>
    </span>
  </NavbarItem__NavAnker>
</div>
`;
